backend/api/Repositories/Cdogs/CdogsRepository.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Pims.Api.Models;
+using Pims.Api.Models.Cdogs;
+
+namespace Pims.Api.Repositories.Cdogs
+{
+    /// <summary>
+    /// CdogsRepository provides document access from a Mayan EDMS api.
+    /// </summary>
+    public class CdogsRepository : CdogsBaseRepository, IDocumentGenerationRepository
+    {
+        private readonly IDocumentGenerationAuthRepository _authRepository;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CdogsRepository"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="httpClientFactory">Injected Httpclient factory.</param>
+        /// <param name="authRepository">Injected repository that handles authentication.</param>
+        /// <param name="configuration">The injected configuration provider.</param>
+        public CdogsRepository(
+            ILogger<CdogsRepository> logger,
+            IHttpClientFactory httpClientFactory,
+            IDocumentGenerationAuthRepository authRepository,
+            IConfiguration configuration)
+            : base(logger, httpClientFactory, configuration)
+        {
+            _authRepository = authRepository;
+        }
+
+        public async Task<ExternalResult<FileTypes>> GetFileTypesAsync()
+        {
+            _logger.LogDebug("Retrieving supported file types...");
+            string authenticationToken = await _authRepository.GetTokenAsync();
```

Comment:
hmm, does it make sense for the repository to ask the other repository for the token or should the service get the token?

Reply:
The auth repository has a 'singleton' instantiation while the other ones just have a 'scope'. That means that the auth repository instance will be re-used during the execution of the application and reused while the other ones get created for each request.

Reply:
Sorry, that wasn't my question. I was wondering if the CdogsRepository should directly get the token from the auth repository or if the service should pass it in.